language/java/language_test.go

```diff
@@ -0,0 +1,220 @@
+package java
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zimmski/osutil"
+	"github.com/zimmski/osutil/bytesutil"
+
+	"github.com/symflower/eval-dev-quality/language"
+	"github.com/symflower/eval-dev-quality/log"
+)
+
+func TestLanguageFiles(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Language *Language
+
+		RepositoryPath string
+
+		ExpectedFilePaths []string
+		ExpectedError     error
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			logOutput, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Log(logOutput.String())
+				}
+			}()
+
+			if tc.Language == nil {
+				tc.Language = &Language{}
+			}
+			actualFilePaths, actualError := tc.Language.Files(logger, tc.RepositoryPath)
+
+			assert.Equal(t, tc.ExpectedFilePaths, actualFilePaths)
+			assert.Equal(t, tc.ExpectedError, actualError)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Plain",
+
+		RepositoryPath: "../../testdata/java/plain",
+
+		ExpectedFilePaths: []string{
+			"src/main/java/com/eval/Plain.java",
+		},
+	})
+}
+
+func TestLanguageImportPath(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Language *Language
+
+		ProjectRootPath string
+		FilePath        string
+
+		ExpectedImportPath string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			if tc.Language == nil {
+				tc.Language = &Language{}
+			}
+
+			actualImportPath := tc.Language.ImportPath(tc.ProjectRootPath, tc.FilePath)
+
+			assert.Equal(t, tc.ExpectedImportPath, actualImportPath)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Source file",
+
+		FilePath: "src/main/java/com/eval/Plain.java",
+
+		ExpectedImportPath: "com.eval",
+	})
+}
+
+func TestLanguageTestFilePath(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Language *Language
+
+		ProjectRootPath string
+		FilePath        string
+
+		ExpectedTestFilePath string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			if tc.Language == nil {
+				tc.Language = &Language{}
+			}
+
+			actualTestFilePath := tc.Language.TestFilePath(tc.ProjectRootPath, tc.FilePath)
+
+			assert.Equal(t, tc.ExpectedTestFilePath, actualTestFilePath)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Source file",
+
+		FilePath: "src/main/java/com/eval/Plain.java",
+
+		ExpectedTestFilePath: "src/test/java/com/eval/PlainTest.java",
+	})
```

Comment:
Would be nice to see also a case where the `src/main/java` is not there. And also one with a root path.